File: packages/google-agones-sdk/bindings/java/src/main/java/com/agones/dev/sdk/AgonesSDK.java
package com.agones.dev.sdk;

import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.function.Consumer;

public interface AgonesSDK {
    void destroy();

    CompletableFuture<GameServer> getGameServer();
    CompletableFuture<String> getState();

    CompletableFuture<Void> setReady();
    CompletableFuture<Void> setAllocated();
    CompletableFuture<Void> setReserved(long seconds);
    void watchGameServer(Consumer<GameServer> consumer);
    void askShutdown();

    void sendHealthcheck();

    Alpha alpha();
    interface Alpha {
        CompletableFuture<Boolean> notifyPlayerConnected(String id);
        CompletableFuture<Boolean> notifyPlayerDisconnected(String id);
        CompletableFuture<Void> setPlayerCapacity(long capacity);
        CompletableFuture<Long> getPlayerCapacity();
        CompletableFuture<Long> getPlayerCount();
        CompletableFuture<Boolean> isPlayerConnected(String id);
        CompletableFuture<List<String>> getConnectedPlayers();
    }
}

File: packages/shulker-agones-java-sdk/src/main/kotlin/agones/dev/sdk/AgonesSDK.kt
package agones.dev.sdk

import agones.dev.sdk.Sdk.GameServer
import java.util.concurrent.CompletableFuture

interface AgonesSDK {
    val alpha: AgonesSDK.Alpha

    fun destroy()

    fun getGameServer(): CompletableFuture<GameServer>
    fun getState(): CompletableFuture<String>

    fun setReady(): CompletableFuture<Unit>
    fun setAllocated(): CompletableFuture<Unit>
    fun setReserved(seconds: Long): CompletableFuture<Unit>
    fun askShutdown()

    fun sendHealthcheck()

    interface Alpha {
        fun notifyPlayerConnected(id: String): CompletableFuture<Boolean>
        fun notifyPlayerDisconnected(id: String): CompletableFuture<Boolean>
        fun setPlayerCapacity(capacity: Long): CompletableFuture<Unit>
        fun getPlayerCapacity(): CompletableFuture<Unit>
        fun getPlayerCount(): CompletableFuture<Unit>
        fun isPlayerConnected(id: String): CompletableFuture<Boolean>
        fun getConnectedPlayers(): CompletableFuture<List<String>>
    }
}
